scanpy/experimental/pp/_normalization.py

```diff
@@ -0,0 +1,243 @@
+    theta
+        The negative binomial overdispersion parameter theta for Pearson residuals.
+        Higher values correspond to less overdispersion (var = mean + mean^2/theta),
+        and `theta=np.Inf` corresponds to a Poisson model.
+    clip
+        Determines if and how residuals are clipped:
+
+            * If `None`, residuals are clipped to the interval [-sqrt(n), sqrt(n)], \
+            where n is the number of cells in the dataset (default behavior).
+            * If any scalar c, residuals are clipped to the interval [-c, c]. Set \
+            `clip=np.Inf` for no clipping.
+
+    n_comps_pca
+        Number of principal components to compute.
+    random_state_pca
+        Change to use different initial states for the optimization.
+    kwargs_pca
+        Dictionary of further keyword arguments passed on to `scanpy.pp.pca()`.
+    use_highly_variable
+        Whether to use the gene selection in `adata.var['highly_variable']` to subset
+        the data before normalizing (default) or proceed on the full dataset.
+    check_values
+        Check if counts in selected layer are integers. A Warning is returned if set to
+        True.
+    inplace
+        Whether to place results in `adata` or return them.
+
+
+    Returns
+    -------
+    If `inplace=False`, returns the Pearson residual-based PCA results
+    (`adata_pca`).
+    If `inplace=True`, updates `adata` with the following fields:
+
+    `.uns['pearson_residuals_normalization']['pearson_residuals_df']`
+         The hvg-subset, normalized by Pearson residuals
+    `.uns['pearson_residuals_normalization']['theta']`
+         The used value of the overdisperion parameter theta
+    `.uns['pearson_residuals_normalization']['clip']`
+         The used value of the clipping parameter
+
+    `.obsm['X_pca']`
+        PCA representation of data after gene selection (if applicable) and Pearson
+        residual normalization.
+    `.uns['pca']['PCs']`
+         The principal components containing the loadings.
+    `.uns['pca']['variance_ratio']`
+         Ratio of explained variance.
+    `.uns['pca']['variance']`
+         Explained variance, equivalent to the eigenvalues of the covariance matrix.
+
+    """
+
+    if use_highly_variable and 'highly_variable' in adata.var_keys():
+        # TODO: are these copies needed?
+        adata_pca = adata[:, adata.var['highly_variable']].copy()
+    else:
+        # TODO: are these copies needed?
+        adata_pca = adata.copy()
+
+    normalize_pearson_residuals(
+        adata_pca, theta=theta, clip=clip, check_values=check_values
+    )
+    pca(adata_pca, n_comps=n_comps_pca, random_state=random_state_pca, **kwargs_pca)
+
+    if inplace:
+        norm_settings = adata_pca.uns['pearson_residuals_normalization']
+        norm_dict = dict(**norm_settings, pearson_residuals_df=adata_pca.to_df())
+        pca_settings = adata_pca.uns['pca']
+        pca_dict = dict(**pca_settings, PCs=adata_pca.varm['PCs'])
```

Comment:
Why is this going into `uns`? Would it make sense for this to be more like how masked genes are handled in the normal PCA function? E.g.:

```python
        if use_highly_variable:
            adata.varm['PCs'] = np.zeros(shape=(adata.n_vars, n_comps))
            adata.varm['PCs'][adata.var['highly_variable']] = pca_.components_.T
        else:
            adata.varm['PCs'] = pca_.components_.T
```

Reply:
adapted output fields and respective tests
(also for the `recipe` function)